File: cpp/Estruturas condicionais/temperatura.cpp
#include <iostream>
using namespace std;

int main() {

    char escalaInicial, escalaFinal;
    double temperatura, resultado;

    cout << "Digite a escala inicial (C/F/K): ";
    cin >> escalaInicial;

    cout << "Digite a escala final (C/F/K): ";
    cin >> escalaFinal;

    cout << "Digite a temperatura: ";
    cin >> temperatura;

    if (escalaInicial == escalaFinal) {
        resultado = temperatura;
    }

    else if ((escalaInicial == 'C' || escalaInicial == 'c') &&
             (escalaFinal == 'F' || escalaFinal == 'f')) {

        resultado = (temperatura * 9.0 / 5.0) + 32.0;
    }

    else if ((escalaInicial == 'C' || escalaInicial == 'c') &&
             (escalaFinal == 'K' || escalaFinal == 'k')) {

        resultado = temperatura + 273.15;
    }

    else if ((escalaInicial == 'F' || escalaInicial == 'f') &&
             (escalaFinal == 'C' || escalaFinal == 'c')) {

        resultado = (temperatura - 32.0) * 5.0 / 9.0;
    }

    else if ((escalaInicial == 'F' || escalaInicial == 'f') &&
             (escalaFinal == 'K' || escalaFinal == 'k')) {

        resultado = (temperatura - 32.0) * 5.0 / 9.0 + 273.15;
    }

    else if ((escalaInicial == 'K' || escalaInicial == 'k') &&
             (escalaFinal == 'C' || escalaFinal == 'c')) {

        resultado = temperatura - 273.15;
    }

    else if ((escalaInicial == 'K' || escalaInicial == 'k') &&
             (escalaFinal == 'F' || escalaFinal == 'f')) {

        resultado = (temperatura - 273.15) * 9.0 / 5.0 + 32.0;
    }

    else {
        cout << "Escalas invalidas!" << endl;
        return 0;
    }

    cout << "Temperatura convertida: " << resultado << " "
         << escalaFinal << endl;

    return 0;
}

File: java/Estruturas condicionais/Temperatura.java
import java.util.Locale;
import java.util.Scanner;

public class Temperatura {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        char origem, destino;
        double temperatura, resultado = 0;

        System.out.print("Digite a escala de origem (C/F/K): ");
        origem = sc.next().toUpperCase().charAt(0);

        System.out.print("Digite a escala de destino (C/F/K): ");
        destino = sc.next().toUpperCase().charAt(0);

        System.out.print("Digite a temperatura: ");
        temperatura = sc.nextDouble();

        if (origem == 'C') {

            if (destino == 'F') {
                resultado = temperatura * 9.0 / 5.0 + 32.0;
            } else if (destino == 'K') {
                resultado = temperatura + 273.15;
            } else {
                resultado = temperatura;
            }

        } else if (origem == 'F') {

            if (destino == 'C') {
                resultado = (temperatura - 32.0) * 5.0 / 9.0;
            } else if (destino == 'K') {
                resultado = (temperatura - 32.0) * 5.0 / 9.0 + 273.15;
            } else {
                resultado = temperatura;
            }

        } else if (origem == 'K') {

            if (destino == 'C') {
                resultado = temperatura - 273.15;
            } else if (destino == 'F') {
                resultado = (temperatura - 273.15) * 9.0 / 5.0 + 32.0;
            } else {
                resultado = temperatura;
            }

        } else {
            System.out.println("Escala de origem inválida.");
            sc.close();
            return;
        }

        System.out.printf("Temperatura convertida: %.2f %c%n", resultado, destino);

        sc.close();
    }
}